# Death Stick Project Analysis
## Electric SUP Propulsion Device - Battery Pack Design

This notebook contains detailed analysis and design calculations for the Death Stick project - an innovative electric propulsion device for surfboard riders.

## Project Overview
- **Project Name**: Death Stick
- **Purpose**: Handheld electric propulsion device combining SUP paddle form factor with electric propeller
- **Key Innovation**: Carbon fiber/aluminum tube (60mm OD) with internal battery pack and propeller end
- **Analysis Focus**: Battery pack optimization and performance calculations

## Device Specifications
- **Construction**: 60mm outer diameter carbon fiber or aluminum tubing
- **Propulsion**: Electric propeller replacing traditional paddle blade
- **Battery**: 10x NMC lithium cells (46mm × 167mm, 31Ah each)
- **Configuration**: Series connection for 37V nominal voltage
- **Total Length**: 1.76m (battery section only)
- **Battery Weight**: 6.8kg

## Analysis Contents
1. **Battery Pack Specifications**: Cell configuration, voltage ranges, capacity calculations
2. **Physical Design Analysis**: Length, weight, and fit calculations within tube constraints
3. **Power Performance**: Current capabilities, continuous and peak power ratings
4. **Energy Storage**: Capacity and runtime analysis (1,147 Wh total)
5. **Design Considerations**: Ergonomics, weight distribution, and usability factors
6. **Performance Summary**: Key specifications and design trade-offs

## Key Results
- **Voltage Range**: 27.0V - 42.0V (discharge to full charge)
- **Energy Storage**: 1.15 kWh capacity
- **Power Output**: 4.6 kW continuous, 11.5 kW peak
- **Runtime**: Extended operation capability for marine use



SyntaxError: invalid syntax (2266180054.py, line 3)

In [21]:
# Import necessary libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime
import warnings
warnings.filterwarnings('ignore')

# Set up plotting style
plt.style.use('seaborn-v0_8')
sns.set_palette("husl")

print("Libraries imported successfully!")
print(f"Analysis started at: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")


Libraries imported successfully!
Analysis started at: 2025-06-22 17:24:43


## Project Description

### Death Stick - Electric SUP Propulsion Device

The "Death Stick" is an innovative electric propulsion device designed for surfboard riders. It combines the familiar form factor of a SUP (Stand-Up Paddleboard) paddle with electric propulsion technology.

**Key Features:**
- **Construction**: Carbon fiber or aluminum tubing (60mm outer diameter)
- **Propulsion**: Electric propeller at the end instead of a traditional paddle
- **Power Source**: Internal battery pack consisting of 10 NMC lithium cells
- **Use Case**: Handheld propulsion aid for surfboard riders

**Battery Pack Design:**
- **Cell Type**: NMC (Nickel Manganese Cobalt) Lithium cylindrical batteries
- **Configuration**: 10 cells in series
- **Cell Dimensions**: 46mm diameter × 167mm length
- **Cell Specifications**: 3.7V nominal, 31Ah capacity
- **Spacing**: 10mm gap between cells
- **Housing**: Fits within 60mm carbon fiber tubing

This analysis focuses on optimizing the battery pack design and overall device performance.

---

## 1. Data Loading and Preprocessing

Load and prepare data for analysis. This section handles:
- Battery pack calculations and specifications
- Design parameter analysis
- Performance modeling


In [10]:
# Battery Pack Specifications and Calculations

# NMC Cell Specifications (from datasheet)
cell_specs = {
    'nominal_voltage': 3.7,  # V
    'max_charge_voltage': 4.2,  # V
    'discharge_cutoff_voltage': 2.7,  # V
    'capacity': 31,  # Ah
    'length': 167,  # mm
    'diameter': 46,  # mm
    'weight': 680,  # g (±10g)
    'internal_resistance': 1,  # mΩ
    'max_discharge_current': 124,  # A (4C)
    'max_peak_discharge': 310,  # A (10C)
    'cycle_life': 1500  # cycles
}

# Battery Pack Configuration
pack_config = {
    'num_cells': 10,
    'cell_spacing': 10,  # mm gap between cells
    'tube_diameter': 60,  # mm outer diameter
    'connection_type': 'series'
}

print("=== DEATH STICK BATTERY PACK ANALYSIS ===\n")

# Calculate total device length (battery section only)
total_battery_length = (pack_config['num_cells'] * cell_specs['length'] + 
                       (pack_config['num_cells'] - 1) * pack_config['cell_spacing'])

print(f"1. DEVICE LENGTH CALCULATIONS:")
print(f"   • Cell length: {cell_specs['length']} mm")
print(f"   • Number of cells: {pack_config['num_cells']}")
print(f"   • Spacing between cells: {pack_config['cell_spacing']} mm")
print(f"   • Total battery pack length: {total_battery_length} mm ({total_battery_length/1000:.2f} m)")

# Calculate voltage specifications
min_pack_voltage = pack_config['num_cells'] * cell_specs['discharge_cutoff_voltage']
nominal_pack_voltage = pack_config['num_cells'] * cell_specs['nominal_voltage']
max_pack_voltage = pack_config['num_cells'] * cell_specs['max_charge_voltage']

print(f"\n2. VOLTAGE SPECIFICATIONS:")
print(f"   • Minimum voltage (discharge cutoff): {min_pack_voltage:.1f} V")
print(f"   • Nominal voltage: {nominal_pack_voltage:.1f} V")
print(f"   • Maximum voltage (fully charged): {max_pack_voltage:.1f} V")

# Calculate capacity (series connection = same capacity as single cell)
pack_capacity = cell_specs['capacity']

print(f"\n3. CAPACITY SPECIFICATIONS:")
print(f"   • Pack capacity: {pack_capacity} Ah")
print(f"   • Energy storage (nominal): {nominal_pack_voltage * pack_capacity:.1f} Wh")

# Calculate total weight
total_weight = pack_config['num_cells'] * cell_specs['weight']

print(f"\n4. WEIGHT AND SIZE:")
print(f"   • Weight per cell: {cell_specs['weight']} g")
print(f"   • Total battery weight: {total_weight} g ({total_weight/1000:.2f} kg)")
print(f"   • Cell diameter: {cell_specs['diameter']} mm (fits in {pack_config['tube_diameter']} mm tube)")

# Current capabilities
max_continuous_current = cell_specs['max_discharge_current']
max_peak_current = cell_specs['max_peak_discharge']

print(f"\n5. CURRENT SPECIFICATIONS:")
print(f"   • Max continuous discharge: {max_continuous_current} A")
print(f"   • Max peak discharge: {max_peak_current} A")
print(f"   • Max continuous power: {nominal_pack_voltage * max_continuous_current:.0f} W")
print(f"   • Max peak power: {nominal_pack_voltage * max_peak_current:.0f} W")


=== DEATH STICK BATTERY PACK ANALYSIS ===

1. DEVICE LENGTH CALCULATIONS:
   • Cell length: 167 mm
   • Number of cells: 10
   • Spacing between cells: 10 mm
   • Total battery pack length: 1760 mm (1.76 m)

2. VOLTAGE SPECIFICATIONS:
   • Minimum voltage (discharge cutoff): 27.0 V
   • Nominal voltage: 37.0 V
   • Maximum voltage (fully charged): 42.0 V

3. CAPACITY SPECIFICATIONS:
   • Pack capacity: 31 Ah
   • Energy storage (nominal): 1147.0 Wh

4. WEIGHT AND SIZE:
   • Weight per cell: 680 g
   • Total battery weight: 6800 g (6.80 kg)
   • Cell diameter: 46 mm (fits in 60 mm tube)

5. CURRENT SPECIFICATIONS:
   • Max continuous discharge: 124 A
   • Max peak discharge: 310 A
   • Max continuous power: 4588 W
   • Max peak power: 11470 W


In [11]:
# Create battery pack data for analysis
battery_data = {
    'cell_id': range(1, pack_config['num_cells'] + 1),
    'position_mm': [i * (cell_specs['length'] + pack_config['cell_spacing']) 
                   for i in range(pack_config['num_cells'])],
    'voltage_min': [cell_specs['discharge_cutoff_voltage']] * pack_config['num_cells'],
    'voltage_nominal': [cell_specs['nominal_voltage']] * pack_config['num_cells'],
    'voltage_max': [cell_specs['max_charge_voltage']] * pack_config['num_cells'],
    'capacity_ah': [cell_specs['capacity']] * pack_config['num_cells'],
    'weight_g': [cell_specs['weight']] * pack_config['num_cells'],
    'length_mm': [cell_specs['length']] * pack_config['num_cells'],
    'diameter_mm': [cell_specs['diameter']] * pack_config['num_cells']
}

df = pd.DataFrame(battery_data)

# Store key calculations for later use
calculations = {
    'total_length_mm': total_battery_length,
    'total_weight_g': total_weight,
    'pack_voltage_min': min_pack_voltage,
    'pack_voltage_nominal': nominal_pack_voltage,
    'pack_voltage_max': max_pack_voltage,
    'pack_capacity_ah': pack_capacity,
    'pack_energy_wh': nominal_pack_voltage * pack_capacity,
    'max_continuous_power_w': nominal_pack_voltage * max_continuous_current,
    'max_peak_power_w': nominal_pack_voltage * max_peak_current
}

print("\nBattery Pack Data Summary:")
print(f"Created dataset with {len(df)} cells")
print(f"Total system: {calculations['total_length_mm']} mm, {calculations['total_weight_g']} g")
print(f"Pack voltage range: {calculations['pack_voltage_min']}-{calculations['pack_voltage_max']} V")
print(f"Energy capacity: {calculations['pack_energy_wh']} Wh")

df.head(pack_config['num_cells'])



Battery Pack Data Summary:
Created dataset with 10 cells
Total system: 1760 mm, 6800 g
Pack voltage range: 27.0-42.0 V
Energy capacity: 1147.0 Wh


cell_id  position_mm  voltage_min  voltage_nominal  voltage_max  \
0        1            0          2.7              3.7          4.2   
1        2          177          2.7              3.7          4.2   
2        3          354          2.7              3.7          4.2   
3        4          531          2.7              3.7          4.2   
4        5          708          2.7              3.7          4.2   
5        6          885          2.7              3.7          4.2   
6        7         1062          2.7              3.7          4.2   
7        8         1239          2.7              3.7          4.2   
8        9         1416          2.7              3.7          4.2   
9       10         1593          2.7              3.7          4.2   

   capacity_ah  weight_g  length_mm  diameter_mm  
0           31       680        167           46  
1           31       680        167           46  
2           31       680        167           46  
3           31       680        167           46  
4           31       680        167           46  
5           31       680        167           46  
6           31       680        167           46  
7           31       680        167           46  
8           31       680        167           46  
9           31       680        167           46

### Key Results Summary

Based on the 10-cell NMC battery pack configuration:

| **Specification** | **Value** |
|-------------------|-----------|
| **Device Length (Battery Section)** | 1,760 mm (1.76 m) |
| **Total Weight (Batteries Only)** | 6.8 kg |
| **Voltage Range** | 27.0 V - 42.0 V |
| **Nominal Voltage** | 37.0 V |
| **Capacity** | 31 Ah |
| **Energy Storage** | 1,147 Wh (1.15 kWh) |
| **Max Continuous Power** | 4,588 W (~4.6 kW) |
| **Max Peak Power** | 11,470 W (~11.5 kW) |

**Design Considerations:**
- The 1.76m battery section is quite long for a handheld device
- 6.8 kg battery weight will require ergonomic consideration
- High power output suitable for marine propulsion
- Significant energy storage for extended operation


---

## 2. Electric Motor Performance Analysis

This section analyzes the compatibility and performance of the selected electric motor with the battery pack design. The analysis covers voltage compatibility, power margins, current capabilities, performance predictions, and overall system assessment.

### Motor Selection: Flipsky 65150 Motor 160KV 3000W

The selected motor is a waterproof (IP68) brushless DC motor with integrated ESC (Electronic Speed Controller), specifically designed for marine applications including surfboards and underwater thrusters.


In [13]:
# Electric Motor Performance Analysis with Battery Pack
# Flipsky 65150 Motor 160KV 3000W IP68 BLDC Motor / ESC 2-in-1 Combo

# Motor Specifications (from datasheet)
motor_specs = {
    'model': 'Flipsky 65150 Motor 160KV 3000W',
    'kv_rating': 160,  # RPM/V
    'voltage_range_s': (4, 13),  # 4S to 13S
    'voltage_range_v': (14.8, 54.6),  # Actual voltage range based on 3.7V nominal per cell
    'rated_current': 30,  # A
    'max_current': 70,  # A
    'rated_power': 1500,  # W
    'max_power': 3000,  # W
    'rated_torque': 2.3,  # N.m
    'max_torque': 4.5,  # N.m
    'weight': 2000,  # g
    'size_diameter': 65,  # mm
    'size_length': 150,  # mm
    'shaft_diameter': 12,  # mm
    'waterproof_rating': 'IP68',
    'built_in_esc': True
}

print("=== MOTOR-BATTERY COMPATIBILITY ANALYSIS ===\n")

# Voltage Compatibility Check
battery_in_motor_range = (calculations['pack_voltage_min'] >= motor_specs['voltage_range_v'][0] and 
                         calculations['pack_voltage_max'] <= motor_specs['voltage_range_v'][1])

print("1. VOLTAGE COMPATIBILITY:")
print(f"   • Motor voltage range: {motor_specs['voltage_range_v'][0]:.1f}V - {motor_specs['voltage_range_v'][1]:.1f}V")
print(f"   • Battery pack range: {calculations['pack_voltage_min']:.1f}V - {calculations['pack_voltage_max']:.1f}V")
print(f"   • Compatibility: {'✓ COMPATIBLE' if battery_in_motor_range else '✗ NOT COMPATIBLE'}")

# Power Analysis
power_supply_capability = min(calculations['max_continuous_power_w'], calculations['max_peak_power_w'])
power_margin_continuous = (calculations['max_continuous_power_w'] / motor_specs['max_power']) * 100
power_margin_peak = (calculations['max_peak_power_w'] / motor_specs['max_power']) * 100

print(f"\n2. POWER ANALYSIS:")
print(f"   • Motor rated power: {motor_specs['rated_power']}W")
print(f"   • Motor maximum power: {motor_specs['max_power']}W")
print(f"   • Battery continuous power: {calculations['max_continuous_power_w']:.0f}W")
print(f"   • Battery peak power: {calculations['max_peak_power_w']:.0f}W")
print(f"   • Power margin (continuous): {power_margin_continuous:.1f}% of motor max")
print(f"   • Power margin (peak): {power_margin_peak:.1f}% of motor max")

# Current Analysis
current_margin_continuous = (max_continuous_current / motor_specs['max_current']) * 100
current_margin_peak = (max_peak_current / motor_specs['max_current']) * 100

print(f"\n3. CURRENT ANALYSIS:")
print(f"   • Motor rated current: {motor_specs['rated_current']}A")
print(f"   • Motor maximum current: {motor_specs['max_current']}A")
print(f"   • Battery continuous current: {max_continuous_current}A")
print(f"   • Battery peak current: {max_peak_current}A")
print(f"   • Current margin (continuous): {current_margin_continuous:.1f}% of motor max")
print(f"   • Current margin (peak): {current_margin_peak:.1f}% of motor max")

# Performance Calculations
# RPM calculations at different voltages
rpm_at_min_voltage = motor_specs['kv_rating'] * calculations['pack_voltage_min']
rpm_at_nominal_voltage = motor_specs['kv_rating'] * calculations['pack_voltage_nominal']
rpm_at_max_voltage = motor_specs['kv_rating'] * calculations['pack_voltage_max']

print(f"\n4. MOTOR PERFORMANCE PREDICTIONS:")
print(f"   • RPM at min voltage ({calculations['pack_voltage_min']:.1f}V): {rpm_at_min_voltage:.0f} RPM")
print(f"   • RPM at nominal voltage ({calculations['pack_voltage_nominal']:.1f}V): {rpm_at_nominal_voltage:.0f} RPM")
print(f"   • RPM at max voltage ({calculations['pack_voltage_max']:.1f}V): {rpm_at_max_voltage:.0f} RPM")

# Weight Analysis
total_system_weight = calculations['total_weight_g'] + motor_specs['weight']
motor_weight_percentage = (motor_specs['weight'] / total_system_weight) * 100
battery_weight_percentage = (calculations['total_weight_g'] / total_system_weight) * 100

print(f"\n5. WEIGHT DISTRIBUTION:")
print(f"   • Battery pack weight: {calculations['total_weight_g']/1000:.1f} kg ({battery_weight_percentage:.1f}%)")
print(f"   • Motor weight: {motor_specs['weight']/1000:.1f} kg ({motor_weight_percentage:.1f}%)")
print(f"   • Total system weight: {total_system_weight/1000:.1f} kg")

# System Efficiency and Runtime Estimates
print(f"\n6. SYSTEM EFFICIENCY ESTIMATES:")

# Estimate runtime at different power levels
runtime_at_rated_power = (calculations['pack_energy_wh'] / motor_specs['rated_power']) * 60  # minutes
runtime_at_max_power = (calculations['pack_energy_wh'] / motor_specs['max_power']) * 60  # minutes

print(f"   • Estimated runtime at rated power ({motor_specs['rated_power']}W): {runtime_at_rated_power:.1f} minutes")
print(f"   • Estimated runtime at max power ({motor_specs['max_power']}W): {runtime_at_max_power:.1f} minutes")

# Create summary dictionary for further analysis
motor_battery_analysis = {
    'voltage_compatible': battery_in_motor_range,
    'power_margin_continuous': power_margin_continuous,
    'power_margin_peak': power_margin_peak,
    'current_margin_continuous': current_margin_continuous,
    'current_margin_peak': current_margin_peak,
    'rpm_range': (rpm_at_min_voltage, rpm_at_max_voltage),
    'total_system_weight_kg': total_system_weight/1000,
    'runtime_rated_min': runtime_at_rated_power,
    'runtime_max_min': runtime_at_max_power
}

print(f"\n7. DESIGN ASSESSMENT:")
if battery_in_motor_range and power_margin_continuous > 100:
    print("   ✓ EXCELLENT COMPATIBILITY - Battery pack exceeds motor requirements")
    print("   ✓ Sufficient power margins for reliable operation")
    print("   ✓ High current capability ensures good acceleration")
elif battery_in_motor_range:
    print("   ✓ GOOD COMPATIBILITY - Battery pack meets motor requirements")
else:
    print("   ✗ COMPATIBILITY ISSUES - Voltage mismatch detected")


=== MOTOR-BATTERY COMPATIBILITY ANALYSIS ===

1. VOLTAGE COMPATIBILITY:
   • Motor voltage range: 14.8V - 54.6V
   • Battery pack range: 27.0V - 42.0V
   • Compatibility: ✓ COMPATIBLE

2. POWER ANALYSIS:
   • Motor rated power: 1500W
   • Motor maximum power: 3000W
   • Battery continuous power: 4588W
   • Battery peak power: 11470W
   • Power margin (continuous): 152.9% of motor max
   • Power margin (peak): 382.3% of motor max

3. CURRENT ANALYSIS:
   • Motor rated current: 30A
   • Motor maximum current: 70A
   • Battery continuous current: 124A
   • Battery peak current: 310A
   • Current margin (continuous): 177.1% of motor max
   • Current margin (peak): 442.9% of motor max

4. MOTOR PERFORMANCE PREDICTIONS:
   • RPM at min voltage (27.0V): 4320 RPM
   • RPM at nominal voltage (37.0V): 5920 RPM
   • RPM at max voltage (42.0V): 6720 RPM

5. WEIGHT DISTRIBUTION:
   • Battery pack weight: 6.8 kg (77.3%)
   • Motor weight: 2.0 kg (22.7%)
   • Total system weight: 8.8 kg

6. SYSTEM EF

---

## 3. Carbon Fiber Tubing Integration Analysis

This section analyzes the carbon fiber tubing specifications and how it integrates with both the battery pack and electric motor. The analysis covers dimensional compatibility, structural requirements, weight distribution, and manufacturing considerations for the complete Death Stick assembly.

### Tubing Selection: Enhanced Composites Roll Wrap Carbon Fiber Tube

The selected tubing is a high-strength carbon fiber tube with unidirectional fibers for stiffness and 90° wrap for hoop strength, finished with 2×2 twill weave and marine-grade epoxy resin system.


In [16]:
# Carbon Fiber Tubing Integration Analysis
# Enhanced Composites Roll Wrap Carbon Fiber Tube - 60 x 56 x 2360 mm

# Carbon Fiber Tubing Specifications
tubing_specs = {
    'model': 'Enhanced Composites Roll Wrap Carbon Fiber Tube',
    'length_mm': 2360,  # mm
    'outer_diameter_mm': 60,  # mm (OD)
    'inner_diameter_mm': 56,  # mm (ID)
    'wall_thickness_mm': 2,  # mm
    'material': '100% Carbon Fiber',
    'resin_system': 'Epoxy',
    'finish': '2x2 Twill Weave',
    'water_absorption': 'Virtually Zero',
    'cost_aud': 517.00,  # AUD including GST
    'marine_suitable': True,
    'cuttable': True,
    'bondable_epoxy': True
}

print("=== CARBON FIBER TUBING INTEGRATION ANALYSIS ===\n")

# Get battery and motor dimensions from previous analyses
battery_length_mm = calculations['total_length_mm']  # 1760 mm
battery_weight_g = calculations['total_weight_g']    # 6800 g
battery_cell_diameter_mm = 46  # mm
motor_diameter_mm = motor_specs['size_diameter']     # 65 mm
motor_length_mm = motor_specs['size_length']         # 150 mm
motor_weight_g = motor_specs['weight']               # 2000 g

print("1. DIMENSIONAL COMPATIBILITY ANALYSIS:")
print(f"   • Tubing available length: {tubing_specs['length_mm']} mm ({tubing_specs['length_mm']/1000:.2f} m)")
print(f"   • Battery pack length: {battery_length_mm} mm ({battery_length_mm/1000:.2f} m)")
print(f"   • Length margin: {tubing_specs['length_mm'] - battery_length_mm} mm ({(tubing_specs['length_mm'] - battery_length_mm)/1000:.2f} m)")

length_utilization = (battery_length_mm / tubing_specs['length_mm']) * 100
print(f"   • Tube length utilization: {length_utilization:.1f}%")

# Internal diameter fit check
fit_clearance = tubing_specs['inner_diameter_mm'] - battery_cell_diameter_mm
print(f"\n   • Tubing inner diameter: {tubing_specs['inner_diameter_mm']} mm")
print(f"   • Battery cell diameter: {battery_cell_diameter_mm} mm")
print(f"   • Radial clearance: {fit_clearance} mm ({fit_clearance/2:.1f} mm per side)")

battery_fits = fit_clearance > 0
print(f"   • Battery fit: {'✓ FITS' if battery_fits else '✗ TOO TIGHT'}")

print(f"\n2. MOTOR MOUNTING ANALYSIS:")
print(f"   • Motor diameter: {motor_diameter_mm} mm")
print(f"   • Tube outer diameter: {tubing_specs['outer_diameter_mm']} mm")
motor_overhang = motor_diameter_mm - tubing_specs['outer_diameter_mm']
print(f"   • Motor diameter overhang: {motor_overhang} mm")

if motor_overhang > 0:
    print(f"   • Mounting solution: External motor mount required")
    print(f"     - Motor clamp/bracket needed")
    print(f"     - Transition from {tubing_specs['outer_diameter_mm']}mm tube to {motor_diameter_mm}mm motor")
else:
    print(f"   • Mounting solution: Motor can be recessed into tube")

print(f"\n3. WEIGHT DISTRIBUTION ANALYSIS:")
total_system_weight_g = battery_weight_g + motor_weight_g
print(f"   • Total system weight: {total_system_weight_g/1000:.1f} kg")

# Estimate carbon fiber tube weight (typical CF tube ~1.5-2.5 g/cm for this size)
# Using approximation: π × ((OD/2)² - (ID/2)²) × length × density
# CF density ≈ 1.6 g/cm³, but tube has resin so ~1.8 g/cm³ effective
import math
tube_volume_cm3 = math.pi * ((tubing_specs['outer_diameter_mm']/20)**2 - (tubing_specs['inner_diameter_mm']/20)**2) * (battery_length_mm/10)
estimated_tube_weight_g = tube_volume_cm3 * 1.8  # g/cm³ effective density of carbon fiber
cut_tube_weight_g = estimated_tube_weight_g * (battery_length_mm / tubing_specs['length_mm'])

print(f"   • Estimated cut tube weight: {cut_tube_weight_g:.0f} g ({cut_tube_weight_g/1000:.2f} kg)")
total_with_tube_weight_g = total_system_weight_g + cut_tube_weight_g
print(f"   • Total system + tube weight: {total_with_tube_weight_g/1000:.1f} kg")

# Weight distribution along length
battery_weight_per_mm = battery_weight_g / battery_length_mm
print(f"   • Battery weight distribution: {battery_weight_per_mm:.2f} g/mm")

print(f"\n4. STRUCTURAL ANALYSIS:")
# Approximate bending calculations (simplified)
tube_wall_area_cm2 = math.pi * ((tubing_specs['outer_diameter_mm']/20)**2 - (tubing_specs['inner_diameter_mm']/20)**2)
print(f"   • Tube wall cross-sectional area: {tube_wall_area_cm2:.2f} cm²")

# Moment of inertia for hollow cylinder
I_moment = math.pi * ((tubing_specs['outer_diameter_mm']/20)**4 - (tubing_specs['inner_diameter_mm']/20)**4) / 64
print(f"   • Second moment of area: {I_moment:.2f} cm⁴")

# Simple cantilevered beam analysis (very approximate)
# Assuming 1m cantilever with total load
cantilever_length_m = 1.0  # Assumed use length
distributed_load_n_per_m = (total_with_tube_weight_g / 1000) * 9.81  # N/m
max_moment_nm = (distributed_load_n_per_m * cantilever_length_m**2) / 2
print(f"   • Est. max bending moment (1m cantilever): {max_moment_nm:.1f} N⋅m")

print(f"\n5. MANUFACTURING & ASSEMBLY REQUIREMENTS:")
cutting_length_mm = battery_length_mm + 50  # Extra 50mm for end caps/mounting
waste_percentage = ((tubing_specs['length_mm'] - cutting_length_mm) / tubing_specs['length_mm']) * 100

print(f"   • Required cutting length: {cutting_length_mm} mm ({cutting_length_mm/1000:.2f} m)")
print(f"   • Material waste: {waste_percentage:.1f}% ({(tubing_specs['length_mm'] - cutting_length_mm)} mm)")
print(f"   • Cutting tools: Fine band saw, hacksaw, or Dremel (as per spec)")
print(f"   • End finishing: Clean cuts, smooth edges required")
print(f"   • Motor mount: Custom bracket/clamp for {motor_overhang}mm overhang")

print(f"\n6. MARINE SUITABILITY:")
print(f"   • Water resistance: {'✓ EXCELLENT' if tubing_specs['water_absorption'] == 'Virtually Zero' else '✗ POOR'}")
print(f"   • Resin system: {tubing_specs['resin_system']} (marine grade)")
print(f"   • Corrosion resistance: ✓ EXCELLENT (carbon fiber)")
print(f"   • UV resistance: ✓ GOOD (clear coat finish)")

print(f"\n7. COST ANALYSIS:")
cost_per_mm = tubing_specs['cost_aud'] / tubing_specs['length_mm']
material_cost_aud = cost_per_mm * cutting_length_mm
print(f"   • Raw tube cost: ${tubing_specs['cost_aud']:.2f} AUD")
print(f"   • Cost per mm: ${cost_per_mm:.3f} AUD/mm")
print(f"   • Material cost for project: ${material_cost_aud:.2f} AUD")
print(f"   • Cost efficiency: {cutting_length_mm/tubing_specs['length_mm']*100:.1f}% utilization")

print(f"\n8. INTEGRATION ASSESSMENT:")
integration_score = 0
assessment_items = []

if battery_fits:
    integration_score += 25
    assessment_items.append("✓ Battery pack fits within tube ID")
else:
    assessment_items.append("✗ Battery pack too large for tube ID")

if tubing_specs['marine_suitable']:
    integration_score += 25
    assessment_items.append("✓ Marine-grade construction suitable for water use")

if tubing_specs['cuttable']:
    integration_score += 20
    assessment_items.append("✓ Tube can be cut to required length")

if length_utilization > 50:
    integration_score += 15
    assessment_items.append(f"✓ Good material utilization ({length_utilization:.1f}%)")
else:
    assessment_items.append(f"⚠ Low material utilization ({length_utilization:.1f}%)")

if motor_overhang <= 10:
    integration_score += 15
    assessment_items.append("✓ Motor mounting manageable")
else:
    integration_score += 5
    assessment_items.append(f"⚠ Motor overhang requires complex mounting ({motor_overhang}mm)")

print(f"   Integration Score: {integration_score}/100")
for item in assessment_items:
    print(f"   {item}")

if integration_score >= 80:
    print(f"\n   🎯 EXCELLENT INTEGRATION - Highly suitable for Death Stick project")
elif integration_score >= 60:
    print(f"\n   ✓ GOOD INTEGRATION - Suitable with minor modifications")
elif integration_score >= 40:
    print(f"\n   ⚠ MODERATE INTEGRATION - Workable but requires significant adaptation")
else:
    print(f"\n   ✗ POOR INTEGRATION - Major design changes needed")

# Store results for potential further analysis
tubing_integration_analysis = {
    'dimensional_fit': battery_fits,
    'length_utilization_pct': length_utilization,
    'motor_mounting_challenge': motor_overhang,
    'estimated_tube_weight_kg': cut_tube_weight_g/1000,
    'total_system_weight_kg': total_with_tube_weight_g/1000,
    'material_cost_aud': material_cost_aud,
    'integration_score': integration_score,
    'marine_suitable': tubing_specs['marine_suitable']
}


=== CARBON FIBER TUBING INTEGRATION ANALYSIS ===

1. DIMENSIONAL COMPATIBILITY ANALYSIS:
   • Tubing available length: 2360 mm (2.36 m)
   • Battery pack length: 1760 mm (1.76 m)
   • Length margin: 600 mm (0.60 m)
   • Tube length utilization: 74.6%

   • Tubing inner diameter: 56 mm
   • Battery cell diameter: 46 mm
   • Radial clearance: 10 mm (5.0 mm per side)
   • Battery fit: ✓ FITS

2. MOTOR MOUNTING ANALYSIS:
   • Motor diameter: 65 mm
   • Tube outer diameter: 60 mm
   • Motor diameter overhang: 5 mm
   • Mounting solution: External motor mount required
     - Motor clamp/bracket needed
     - Transition from 60mm tube to 65mm motor

3. WEIGHT DISTRIBUTION ANALYSIS:
   • Total system weight: 8.8 kg
   • Estimated cut tube weight: 861 g (0.86 kg)
   • Total system + tube weight: 9.7 kg
   • Battery weight distribution: 3.86 g/mm

4. STRUCTURAL ANALYSIS:
   • Tube wall cross-sectional area: 3.64 cm²
   • Second moment of area: 0.96 cm⁴
   • Est. max bending moment (1m cantilever